app/controllers/review_bids_controller.rb

```diff
@@ -96,26 +103,49 @@ def set_priority
   def assign_bidding
     # sets parameters used for running bidding algorithm
     assignment_id = params[:assignment_id].to_i
+    assignment = Assignment.find(assignment_id)
     # list of reviewer id's from a specific assignment
-    reviewer_ids = AssignmentParticipant.where(parent_id: assignment_id).ids
-    bidding_data = ReviewBid.bidding_data(assignment_id, reviewer_ids)
-    matched_topics = run_bidding_algorithm(bidding_data)
-    ReviewBid.assign_review_topics(assignment_id, reviewer_ids, matched_topics)
-    Assignment.find(assignment_id).update(can_choose_topic_to_review: false) # turns off bidding for students
+    reviewer_ids = assignment.participants.pluck(:id)
+    bidding_data = assignment.review_bid.bidding_data(assignment_id, reviewer_ids)
+    puts bidding_data
+    matched_topics = run_bidding_algorithm(bidding_data,reviewer_ids, assignment_id)
+    puts matched_topics
+    assignment.review_bid.assign_review_topics(assignment_id, reviewer_ids, matched_topics)
+    assignment.update(can_choose_topic_to_review: false) # turns off bidding for students
     redirect_back fallback_location: root_path
   end
+  
 
   # call webserver for running assigning algorithm
   # passing webserver: student_ids, topic_ids, student_preferences, time_stamps
   # webserver returns:
   # returns matched assignments as json body
-  def run_bidding_algorithm(bidding_data)
-    # begin
-    url = 'http://app-csc517.herokuapp.com/match_topics' # hard coding for the time being
-    response = RestClient.post url, bidding_data.to_json, content_type: 'application/json', accept: :json
-    JSON.parse(response.body)
-  rescue StandardError
-    false
-    # end
+  def run_bidding_algorithm(bidding_data, reviewer_ids, assignment_id)
+    
+    #url = WEBSERVICE_CONFIG["review_bidding_webservice_url"] #won't work unless ENV variables are configured
+    begin  
+      #response = RestClient.post(url, bidding_data.to_json, content_type: 'application/json', accept: :json)
+      #matched_topics= JSON.parse(bidding_data)
+      topic_ids_with_team = SignedUpTeam.where.not(team_id: nil).pluck(:topic_id) #Topics which have been selected by teams for submission
+      topics = SignUpTopic.where(assignment_id: assignment_id, id: topic_ids_with_team)
+      
+      unbidded_users = bidding_data["users"].select { |user_id, details| details["tid"].empty? }.keys
+      bidded_users= reviewer_ids - unbidded_users
+      matched_topics = {}
+      bidded_users.each do |reviewer_id|
+        matched_topics[reviewer_id] = bidding_data["users"][reviewer_id]["tid"]
+      end
+      unbidded_users.each do |reviewer_id|
+        # Randomly select 2 distinct topics for this reviewer. Ensuring we have unique topics if possible.
```

Comment:
Why 2 topics?  Shouldn't it depend on the number of reviews required for this assignment?  Sounds like someone is hard-wiring a policy that needs to use a value set elsewhere.